core/authenticate/session/service.go

```diff
@@ -8,30 +8,38 @@ import (
 	"github.com/odpf/shield/internal/server/consts"
 
 	"github.com/google/uuid"
+	"github.com/odpf/salt/log"
+	"github.com/robfig/cron/v3"
 	"google.golang.org/grpc/metadata"
 )
 
 var (
 	ErrNoSession       = errors.New("no session")
 	ErrDeletingSession = errors.New("error deleting session")
+	refreshTime        = "0 0 * * *" // Once a day at midnight (UTC)
 )
 
 type Repository interface {
 	Set(ctx context.Context, session *Session) error
 	Get(ctx context.Context, id uuid.UUID) (*Session, error)
 	Delete(ctx context.Context, id uuid.UUID) error
+	DeleteExpiredSessions(ctx context.Context, logger log.Logger, expiryTime time.Time) error
```

Comment:
This method doesn't look appropriate to me. Session already have an expiry time and using this method we are passing time of `now` which tells it to expire all sessions past `now`. I don't think we would ever call this method without `now` time so making it an argument doesn't look good.

I feel it's better if the repository itself has a `Now` method to get current time and simply delete all sessions older than expiry time. 

Same in `flows` as well. Either we can add another column in flow with expiry_at just like session or we can initialize our postgres repository with validity duration. Then Cleanup method will simply do a Delete call where `created_at < now - validity`.

Reply:
@kushsharma you pointed this out last time, that its better if we pass our expiry time manually if in case db timezone is not matching the application's timezone. That's why this expiryTime argument was added